### DISCUSSION TITLE
Array with empty/absent `items` generates untyped `List`

### DISCUSSION THREAD

**Discussion by oefe** [koxudaxi/datamodel-code-generator#1809] [Q&A] [answered]
I'm not sure whether this would be considered a bug, but an OpenAPI array with empty `items` (or no `items` field at all), like this:

```json
          "contents": {
            "type": "array",
            "items": { },
            "nullable": true
          }
```

Generates a list without type arguments:
```python
    contents: Optional[List] = None
```

While this is syntactically correct, type checkers may warn about this.
For example Pylance will issue [reportMissingTypeArgument](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportMissingTypeArgument):

> Expected type arguments for generic class "list"

I would therefore expect a type argument:
```python
    contents: Optional[List[Any]] = None
```

**Comment by ilovelinux** [comment, marked as answer]
Good news!

- Related to https://github.com/koxudaxi/datamodel-code-generator/issues/2325
- Fixed by #2584 

Closing :slightly_smiling_face:
